src/data/fintype/basic.lean

```diff
@@ -478,45 +478,65 @@ by simp [finset.subset_iff, set.subset_def]
-  [fintype s] [fintype t] : (s ∩ t).to_finset = s.to_finset ∩ t.to_finset :=
+lemma to_finset_inter [decidable_eq α] (s t : set α) [fintype s] [fintype t] [fintype ↥(s ∩ t)] :
+  (s ∩ t).to_finset = s.to_finset ∩ t.to_finset :=
 by { ext, simp }
 
-lemma to_finset_union {α : Type*} [decidable_eq α] (s t : set α) [fintype (s ∪ t : set α)]
-  [fintype s] [fintype t] : (s ∪ t).to_finset = s.to_finset ∪ t.to_finset :=
+lemma to_finset_union [decidable_eq α] (s t : set α) [fintype s] [fintype t] [fintype ↥(s ∪ t)] :
+  (s ∪ t).to_finset = s.to_finset ∪ t.to_finset :=
 by { ext, simp }
 
-lemma to_finset_diff {α : Type*} [decidable_eq α] (s t : set α) [fintype s] [fintype t]
-  [fintype (s \ t : set α)] : (s \ t).to_finset = s.to_finset \ t.to_finset :=
+lemma to_finset_diff [decidable_eq α] (s t : set α) [fintype s] [fintype t] [fintype ↥(s \ t)] :
+  (s \ t).to_finset = s.to_finset \ t.to_finset :=
 by { ext, simp }
 
-lemma to_finset_ne_eq_erase {α : Type*} [decidable_eq α] [fintype α] (a : α)
-  [fintype {x : α | x ≠ a}] : {x : α | x ≠ a}.to_finset = finset.univ.erase a :=
-by { ext, simp }
+lemma to_finset_symm_diff [decidable_eq α] [fintype s] [fintype t] [fintype ↥(s ∆ t)] :
+  (s ∆ t).to_finset = s.to_finset ∆ t.to_finset :=
+by { ext, simp [mem_symm_diff, finset.mem_symm_diff] }
 
-theorem to_finset_compl [decidable_eq α] [fintype α] (s : set α) [fintype s] [fintype ↥sᶜ] :
-  (sᶜ).to_finset = s.to_finsetᶜ :=
+lemma to_finset_compl [decidable_eq α] [fintype α] (s : set α) [fintype s] [fintype ↥sᶜ] :
+  sᶜ.to_finset = s.to_finsetᶜ :=
 by { ext, simp }
 
-@[simp] lemma to_finset_eq_univ [fintype α] {s : set α} [fintype s] :
-  s.to_finset = finset.univ ↔ s = set.univ :=
-by rw [← coe_inj, coe_to_finset, coe_univ]
+/- TODO The `↥` circumvents an elaboration bug. See comment on `set.to_finset_univ`. -/
+@[simp] lemma to_finset_empty [fintype ↥(∅ : set α)] : (∅ : set α).to_finset = ∅ := by { ext, simp }
 
-/- TODO Without the coercion arrow (`↥`) there is an elaboration bug;
+/- TODO Without the coercion arrow (`↥`) there is an elaboration bug in the following two;
 it essentially infers `fintype.{v} (set.univ.{u} : set α)` with `v` and `u` distinct.
 Reported in leanprover-community/lean#672 -/
-@[simp] lemma to_finset_univ [fintype ↥(set.univ : set α)] [fintype α] :
+@[simp] lemma to_finset_univ [fintype α] [fintype ↥(set.univ : set α)] :
   (set.univ : set α).to_finset = finset.univ :=
-to_finset_eq_univ.2 rfl
+by { ext, simp }
+
+@[simp] lemma to_finset_eq_empty [fintype s] : s.to_finset = ∅ ↔ s = ∅ :=
+by rw [←to_finset_empty, to_finset_inj]
+
+@[simp] lemma to_finset_eq_univ [fintype α] [fintype s] : s.to_finset = finset.univ ↔ s = univ :=
+by rw [← coe_inj, coe_to_finset, coe_univ]
+
+lemma to_finset_ne_eq_erase {α : Type*} [decidable_eq α] [fintype α] (a : α)
+  [fintype {x : α | x ≠ a}] : {x : α | x ≠ a}.to_finset = finset.univ.erase a :=
+by { ext, simp }
```

Comment:
Do we have `to_finset_set_of` that translates to `finset.univ.filter`? Then `filter_ne` would handle this.

Reply:
That sounds like a much better lemma indeed.